invenio_app_ils/permissions.py

```diff
@@ -79,6 +90,24 @@ def files_permission(obj, action=None):
     return backoffice_permission()
 
 
+def is_loan_owner_permission(patron_pid):
+    """Return permission to allow loan owner, librarians and admins."""
+    return Permission(UserNeed(int(patron_pid)), backoffice_access_action)
+
+
+def loan_extend_circulation_permission(loan):
+    """Return permission to allow only owner and librarians to extend loan."""
+    if not current_user or not current_user.id:
+        abort(401)
+
+    if current_user.id == int(loan["patron_pid"]):
+        loan = loan.replace_refs()
+        if loan.get("document").get("circulation").get("overbooked"):
+            raise InvalidLoanExtendError("This document is overbooked!")
+
+    return is_loan_owner_permission(loan["patron_pid"])
```

Comment:
```suggestion
    return LoanOwnerPermission(loan)
```

and remove `is_loan_owner_permission`